### ISSUE TITLE
Benchmarking daggerhashimoto problem

### ISSUE THREAD

**Issue by Simaex** [nicehash/NiceHashMiner#791]
I was trying to understand what is causing benchmarking issues for daggerhashimoto (solo and dualmining) that were reported by me and other users before. At last I understand that we are benchmarking algo that can connect to different pools for mining actual coins - Ethereum | Ethereum Classic | Musicoin | Expanse | Ubiq | Soilcoin that have different DAG size. When running miner it is shown by selecting different EPOCH from time to time but for single coin EPOCH is changed only once in 30k blocks. It is resulting in big difference of actual hashrate (in my case - up to three times) that can't be controlled by user running NHML. Manual settings are no good - either we set high value and get poor profitability or we set low value and actually forbid the algo. Let's try to figure what can be done. 

#781 
#644 
#486 
Also connected to #611 #790

**Comment by SNiCKeeRS** [comment]
so there is no problem? Is the constant total speed change normal in dual mining? #790

**Comment by Simaex** [comment]
@SNiCKeeRS I feel that we have problem. Speed change when randomly changing EPOCH is causing ineffective running for the algo.

**Comment by SNiCKeeRS** [comment]
@Simaex  do you have a suggestion on what we should do?

**Comment by Simaex** [comment]
@SNiCKeeRS no useful ideas up to date. The only workaround I am able to suggest is to set a watchdog that will switch algo when speed is not as benchmarked and to set some kind of offset timer for a failed algo to prevent returning to it frequently.

**Comment by SNiCKeeRS** [comment]
I fixed the problem by changing the operating system. I have converted the operating system from Windows 10 to Windows 8.1. In the future, it will help those who get the error.

**Comment by S74nk0** [comment]
Issue related to old nhm version/build. Please use latest version and open a new issue for bugs and/or features.
